services/chaospod_test.go

```diff
@@ -202,26 +203,25 @@ var _ = Describe("Chaos Pod Service", func() {
 						Expect(chaosPods).Should(HaveLen(len(chaosPodsObjects)))
 					})
 				})
-			})
 
-			Describe("failed cases", func() {
-				When("the k8s client return an error", func() {
+				Describe("failed cases", func() {
+					When("the k8s client return an error", func() {
 
-					BeforeEach(func() {
-						// Arrange
-						clientMock := mocks.NewK8SClientMock(GinkgoT())
-						clientMock.EXPECT().List(mock.Anything, mock.Anything, mock.Anything).Return(fmt.Errorf("error"))
-						chaosPodServiceConfig.Client = clientMock
-					})
+						BeforeEach(func() {
+							// Arrange
+							k8sClientMock.EXPECT().List(mock.Anything, mock.Anything, mock.Anything).Return(fmt.Errorf("error"))
+							chaosPodServiceConfig.Client = k8sClientMock
```

Comment:
Is this necessary to re-assign the mock here? We are already assigning it in the first BeforeEach 🤔